Mathlib/RingTheory/UniqueFactorizationDomain/Defs.lean

```diff
@@ -197,3 +197,8 @@ theorem irreducible_of_factor {a : α} : ∀ x : α, x ∈ factors a → Irreduc
   (prime_of_factor x h).irreducible
 
 end UniqueFactorizationMonoid
+
+lemma Irreducible.unit_pow_iff {M : Type*} [Monoid M] {x : M} (hx : Irreducible x) {n : ℕ} :
+    IsUnit (x ^ n) ↔ n = 0 := by
+  rcases n with (_|n) <;>
+  simp [hx.not_unit]
```

Comment:
Not here, I think :)

Can you get away with putting
```
lemma isUnit_pow_iff_of_not_isUnit {M : Type*} [Monoid M] {x : M} (hx : ¬ IsUnit x) {n : ℕ} :
    IsUnit (x ^ n) ↔ n = 0 := by
  rcases n with (_|n) <;>
  simp [hx]
```
in `Mathlib.Algebra.Group.Commute.Units` with the similar theorems and using `isUnit_pow_iff_of_not_isUnit hϖ.not_unit` in your application?

(If you want to keep this, it should certainly have `isUnit` in the name rather than `unit`)